docs/analytics-gtm/gtm-link-click-tracker.md

```diff
@@ -0,0 +1,62 @@
+# Google Tag Manager link click tracker
+
+A module exists for tracking link clicks on GOV.UK.
+
+When initialised, the JS adds an event listener to the `<body>` of the page with the following listeners:
+
+- A `click` listener (for left clicks, control clicks, and "Meta key" clicks - "Meta key" clicks are command + clicks or windows key + clicks.)
+- A `mousedown` listener (for middle clicks.)
+- A `contextmenu` listener (for right clicks / context menu keypresses.)
+
+When one of these listeners are fired, they check if the `event.target` is an `<a>` tag with a `href`. If so, it will then check if the link is:
+
+- A `download` link - This is either a href to `assets.publishing.service.gov.uk` or a href on the `www.gov.uk` domain pointing to the `/government/uploads/` path.
+- A `mailto` link - This is an email href that starts with `mailto:`.
+- An `external` link - This is a href that is not on the `www.gov.uk` domain. 
+    - To calculate this, it checks if the href starts with `http://www.gov.uk/`, `https://www.gov.uk/`, `//www.gov.uk/`, `www.gov.uk/`, `http://gov.uk/`, `https://gov.uk/`, `//gov.uk`, or `gov.uk/` (NB: `gov.uk/` and `www./` are technically relative links as a href must contain a `/` or http method at the start to be treated as non-relative).
+    - It also checks if a `href` starts with `/` and does not start with `//` to determine if it's a relative link like `/bank-holidays` , but not a protocol relative link `//google.com`
+    - If the `href` does not meet both of these criteria, then it is considered an external link.
+
+## Basic use
+
+```
+//= require ./analytics-ga4/gtm-schemas
+//= require ./analytics-ga4/gtm-link-click-tracker
+
+window.GOVUK.analyticsGA4.linkClickTracker.trackLinkClicks()
+```
```

Comment:
I'm not sure we need to document how to include the script, as we'll be including the script in GOV.UK anyway. Also we won't be including it in this way (once we integrate it into the cookies code) so I wouldn't want someone unfamiliar with the analytics code copying and pasting this into an application.